src/main.cpp

```diff
@@ -2511,12 +2502,17 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             return state.DoS(100, error("ConnectBlock(): too many sigops"),
                              REJECT_INVALID, "bad-blk-sigops");
 
+
```

Comment:
These two calls could be moved in the block below since both of them return true in case of coin base tx

Reply:
On the other hand the IsCoinBase case is explicitly handled into the functions [I added it for HaveDependencies method], hence we could as well leave them outside the block. This way the IsCoinBase check is done once (while moving into the loop it'll be done twice). 
Also I find it somewhat more readable outside the loop. Don't you too?

Reply:
Agreed